### DISCUSSION TITLE
Trainer(amp_level='O2')

### DISCUSSION THREAD

**Discussion by RuixiangZhao** [Lightning-AI/pytorch-lightning#10950] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
When I try to set `Trainer` like this:

```python
trainer = Trainer(
    num_sanity_val_steps=0, 
    logger=cfg.load_loggers,
    callbacks=cfg.callbacks,
    max_epochs= cfg.General.epochs,
    gpus=cfg.General.gpus,
    amp_level=cfg.General.amp_level, # O2
    precision=cfg.General.precision,  
    accumulate_grad_batches=cfg.General.grad_acc,
    deterministic=True,
    check_val_every_n_epoch=1,
)
```

It will throw an error:

`pytorch_lightning.utilities.exceptions.MisconfigurationException: You have asked for amp_level='O2' but it's only supported with amp_backend='apex'`

**Comment by tchaton** [comment, marked as answer]
Hey @RuixiangZhao,

There are currently 2 precision backends. AMP and APEX. level are supported only with apex and you need to provide Trainer(amp_backend='apex') to activate it as native is the default.

I am curious, what was missing from the `MisconfigurationException` to make this clearer?
